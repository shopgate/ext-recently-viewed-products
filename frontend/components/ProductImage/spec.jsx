import React from 'react';
import { mount } from 'enzyme';
import ProductImage from './index';

describe('ProductImage', () => {
  it('should render an image', () => {
    const component = mount(<ProductImage alt="Foo" src="http://localhost" />);
    expect(component.find('Image').exists()).toBe(true);
    expect(component.find('Placeholder').exists()).toBe(false);
    expect(component).toMatchSnapshot();
  });
  it('should render a placeholder and then an image', () => {
    const component = mount(<ProductImage alt="Foo" src={null} />);
    expect(component.find('Image').exists()).toBe(false);
    expect(component.find('Placeholder').exists()).toBe(true);
    expect(component).toMatchSnapshot();
    component.setProps({ src: 'http://localhost' });
    component.update();
    expect(component.find('Image').exists()).toBe(true);
    expect(component.find('Placeholder').exists()).toBe(false);
    expect(component).toMatchSnapshot();
  });
  it('should render an image and then placeholder on error', () => {
    const component = mount(<ProductImage alt="Foo" src="http://localhost" />);
    expect(component.find('Image').exists()).toBe(true);
    expect(component.find('Placeholder').exists()).toBe(false);
    expect(component).toMatchSnapshot();
    component.at(0).instance().imageLoadingFailed();
    component.update();
    expect(component.find('Image').exists()).toBe(false);
    expect(component.find('Placeholder').exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
