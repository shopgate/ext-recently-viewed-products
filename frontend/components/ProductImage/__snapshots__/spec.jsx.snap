// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should render a placeholder and then an image 1`] = `
<ProductImage
  alt="Foo"
  animating={true}
  forcePlaceholder={false}
  highestResolutionLoaded={[Function]}
  ratio={null}
  resolutions={
    Array [
      Object {
        "blur": 2,
        "height": 50,
        "width": 50,
      },
      Object {
        "height": 440,
        "width": 440,
      },
    ]
  }
  src={null}
>
  <div
    className="css-y5kkpq"
  >
    <div
      className="css-15j45j0"
    >
      <Placeholder
        className="css-19rc2oq"
      >
        <Icon
          className="css-19rc2oq"
          color={null}
          content="<circle cx=\\"12\\" cy=\\"12\\" r=\\"3.2\\"/><path d=\\"M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z\\"/><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"/>"
          size="inherit"
          viewBox="0 0 24 24"
        >
          <svg
            className="css-yys9hb css-19rc2oq"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<circle cx=\\"12\\" cy=\\"12\\" r=\\"3.2\\"/><path d=\\"M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z\\"/><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"/>",
              }
            }
            style={
              Object {
                "fill": null,
                "fontSize": "inherit",
              }
            }
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          />
        </Icon>
      </Placeholder>
    </div>
  </div>
</ProductImage>
`;

exports[`ProductImage should render a placeholder and then an image 2`] = `
<ProductImage
  alt="Foo"
  animating={true}
  forcePlaceholder={false}
  highestResolutionLoaded={[Function]}
  ratio={null}
  resolutions={
    Array [
      Object {
        "blur": 2,
        "height": 50,
        "width": 50,
      },
      Object {
        "height": 440,
        "width": 440,
      },
    ]
  }
  src="http://localhost"
>
  <Image
    alt="Foo"
    animating={true}
    backgroundColor="#f2f2f2"
    className=""
    forcePlaceholder={false}
    highestResolutionLoaded={[Function]}
    onError={[Function]}
    onLoad={null}
    ratio={null}
    resolutions={
      Array [
        Object {
          "blur": 2,
          "height": 50,
          "width": 50,
        },
        Object {
          "height": 440,
          "width": 440,
        },
      ]
    }
    src="http://localhost"
    srcmap={null}
    transition={null}
  >
    <div
      className="css-vfdozt "
    />
  </Image>
</ProductImage>
`;

exports[`ProductImage should render an image 1`] = `
<ProductImage
  alt="Foo"
  animating={true}
  forcePlaceholder={false}
  highestResolutionLoaded={[Function]}
  ratio={null}
  resolutions={
    Array [
      Object {
        "blur": 2,
        "height": 50,
        "width": 50,
      },
      Object {
        "height": 440,
        "width": 440,
      },
    ]
  }
  src="http://localhost"
>
  <Image
    alt="Foo"
    animating={true}
    backgroundColor="#f2f2f2"
    className=""
    forcePlaceholder={false}
    highestResolutionLoaded={[Function]}
    onError={[Function]}
    onLoad={null}
    ratio={null}
    resolutions={
      Array [
        Object {
          "blur": 2,
          "height": 50,
          "width": 50,
        },
        Object {
          "height": 440,
          "width": 440,
        },
      ]
    }
    src="http://localhost"
    srcmap={null}
    transition={null}
  >
    <div
      className="css-vfdozt "
    />
  </Image>
</ProductImage>
`;

exports[`ProductImage should render an image and then placeholder on error 1`] = `
<ProductImage
  alt="Foo"
  animating={true}
  forcePlaceholder={false}
  highestResolutionLoaded={[Function]}
  ratio={null}
  resolutions={
    Array [
      Object {
        "blur": 2,
        "height": 50,
        "width": 50,
      },
      Object {
        "height": 440,
        "width": 440,
      },
    ]
  }
  src="http://localhost"
>
  <Image
    alt="Foo"
    animating={true}
    backgroundColor="#f2f2f2"
    className=""
    forcePlaceholder={false}
    highestResolutionLoaded={[Function]}
    onError={[Function]}
    onLoad={null}
    ratio={null}
    resolutions={
      Array [
        Object {
          "blur": 2,
          "height": 50,
          "width": 50,
        },
        Object {
          "height": 440,
          "width": 440,
        },
      ]
    }
    src="http://localhost"
    srcmap={null}
    transition={null}
  >
    <div
      className="css-vfdozt "
    />
  </Image>
</ProductImage>
`;

exports[`ProductImage should render an image and then placeholder on error 2`] = `
<ProductImage
  alt="Foo"
  animating={true}
  forcePlaceholder={false}
  highestResolutionLoaded={[Function]}
  ratio={null}
  resolutions={
    Array [
      Object {
        "blur": 2,
        "height": 50,
        "width": 50,
      },
      Object {
        "height": 440,
        "width": 440,
      },
    ]
  }
  src="http://localhost"
>
  <div
    className="css-y5kkpq"
  >
    <div
      className="css-15j45j0"
    >
      <Placeholder
        className="css-19rc2oq"
      >
        <Icon
          className="css-19rc2oq"
          color={null}
          content="<circle cx=\\"12\\" cy=\\"12\\" r=\\"3.2\\"/><path d=\\"M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z\\"/><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"/>"
          size="inherit"
          viewBox="0 0 24 24"
        >
          <svg
            className="css-yys9hb css-19rc2oq"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<circle cx=\\"12\\" cy=\\"12\\" r=\\"3.2\\"/><path d=\\"M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z\\"/><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"/>",
              }
            }
            style={
              Object {
                "fill": null,
                "fontSize": "inherit",
              }
            }
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          />
        </Icon>
      </Placeholder>
    </div>
  </div>
</ProductImage>
`;
